src/StructuredLogger.Tests/BinaryLogTests.cs

```diff
@@ -0,0 +1,164 @@
+using Microsoft.Build.Logging.StructuredLogger;
+using Microsoft.Build.Logging.StructuredLogger.UnitTests;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Build.Logging.StructuredLogger.UnitTests
+{
+    /// <summary>
+    /// Unit tests for the <see cref = "BinaryLog"/> class.
+    /// </summary>
+    public class BinaryLogTests
+    {
+        /// <summary>
+        /// Tests that calling ReadRecords with a null string argument throws an exception.
+        /// </summary>
+        [Fact]
+        public void ReadRecords_String_NullInput_ThrowsException()
+        {
+            // Arrange
+            string nullPath = null;
+            // Act & Assert
+            Assert.ThrowsAny<Exception>(() => BinaryLog.ReadRecords(nullPath));
+        }
+
+        /// <summary>
+        /// Tests that calling ReadRecords with a non-existent file path throws an exception.
+        /// </summary>
+        [Fact]
+        public void ReadRecords_String_NonExistentFile_ThrowsException()

```

Comment:
This is a duplicate of `ReadBuild_String_NonExistentFile_ThrowsFileNotFoundException` with less specific exception - so it's superfluous